packages/data-transport-layer/src/db/transport-db.ts

```diff
@@ -379,38 +379,14 @@ export class TransportDB {
     if (index === null) {
       return null
     }
-    let entry = await this.db.get<TEntry>(`${key}:index`, index)
-    entry = stringify(entry)
-    return entry
+    return this.db.get<TEntry>(`${key}:index`, index)
   }
 
   private async _getEntries<TEntry extends Indexed>(
     key: string,
     startIndex: number,
     endIndex: number
   ): Promise<TEntry[] | []> {
-    const entries = await this.db.range<TEntry>(
-      `${key}:index`,
-      startIndex,
-      endIndex
-    )
-    const results = []
-    for (const entry of entries) {
-      results.push(stringify(entry))
-    }
-    return results
-  }
-}
-
-function stringify(entry) {
-  if (entry === null || entry === undefined) {
-    return entry
-  }
-  if (entry.gasLimit) {
-    entry.gasLimit = BigNumber.from(entry.gasLimit).toString()
-  }
-  if (entry.decoded) {
-    entry.decoded.gasLimit = BigNumber.from(entry.decoded.gasLimit).toString()
+    return this.db.range<TEntry>(`${key}:index`, startIndex, endIndex)
```

Comment:
Got confused here but realized that this was meant to target `regenesis/0.4.0`. Which is perfectly fine. LGTM!